### ISSUE TITLE
after upgrade theme setting is gone

### ISSUE THREAD

**Issue by mcnesium** [nextcloud/server#2785]
After upgrading from 10.0.1 to 10.0.2 theme setting in `config.php` has been set to `  'theme' => '',`

(reopened from [here](https://github.com/nextcloud/updater/issues/71))

**Comment by MorrisJobke** [comment]
cc @LukasReschke @rullzer @karlitschek

**Comment by nickvergessen** [comment]
Well this is still the intended behaviour. Themes are disabled on updates because we don't know whether they have been adjusted to the new changes yet?

**Comment by nickvergessen** [comment]
https://github.com/nextcloud/server/blob/30689f338faf5327e13b65d3b4a2d85f9e79064d/lib/base.php#L377-L378

It also prints the message:
> The theme %s has been disabled.

I'm not really sure if we should risk this.

**Comment by mcnesium** [comment]
I understand your approach. Though, compared to other software, it seems a little strange to me as an administrator, that an nc instance is more or less reset to default after an upgrade. That includes not only the theme, but apps as well. Any non-default app is disabled after an upgrade. I only use calendar, contacts, direct_menu and mail, so I can't tell whether this is true for all non-default apps.

While after using and upgrading oc/nc for years I am used to just `occ app:enable foo` by now after an upgrade, I was not really aware of themes being reset too, even though there is a note as you state.

So there are two enhancements coming to my mind:
* There should be an occ command to (re-)enable a theme just like an app
* After an upgrade, disabled features should be announced more obvious, both in the browser and on command line. For example a red-backgrounded note "Congratulations, everything went fine, BUT: app 1, app 2, theme X … are still disabled. Re-enable them using `occ re-enable everything`"

**Comment by juliushaertl** [comment]
@mcnesium `./occ config:system:set theme --value=MyTheme` should work just fine for enabling the theme again. :wink:

**Comment by nickvergessen** [comment]
I think the console is already coloring it good:
![bildschirmfoto vom 2017-01-18 10-59-06](https://cloud.githubusercontent.com/assets/213943/22059506/52c4cb9e-dd6d-11e6-8da0-3d86f3cfb1f2.png)


But yeah, the web UI needs a bit more clearness on that.

**Comment by mcnesium** [comment]
This notice scrolls up and is gone, if there is more output. Not unlikely on a production setting. Thats why I proposed to announce it at the end of the upgrade process again.

**Comment by DJCrashdummy** [comment]
**this also appeared after the update of just the calendar-app itself!**

less than a week ago calender 1.5.2 was published and i installed it right away. after this the internal/server-update with the original background-image _(i don't know how it is called correct)_ had to be done. everything went smoothly and so i didn't think further about it...
then a user hinted me that there is a other theme present and i had to detect that `'theme' => 'mytheme',` had changed to `'theme' => '',`!

i understand and fine with it if themes are disabled for safety reasons, but it would be nice to restore the used theme afterwards or at least a hint after the update to manually do so.

**Comment by szaimen** [comment]
I'm closing this issue due to inactivity. If this is still happening please make sure to **upgrade** to the **latest** version. After that, feel free to reopen.

**Comment by mcnesium** [comment]
Yes, it is still happening. Just updated the official Docker image from 21.0.1 to 21.0.2

[My comment over there](https://github.com/nextcloud/desktop/issues/1110#issuecomment-845846092) applies here, too :raised_eyebrow:

**Comment by szaimen** [comment]
Sorry but we are currently trying to get rid of stale issues. Thats why I've asked to reopen thiss issue if it should still fail.
